.github/workflows/test.yaml

```
@@ -0,0 +1,35 @@
+name: Pull Request
+on:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  provide_modules:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(find modules -mindepth 1 -maxdepth 1 -type d | jq -R | jq -sc)"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  validate:
+    needs: provide_modules
+    strategy:
+      matrix:
+        module: ${{ fromJson(needs.provide_modules.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ matrix.module }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Extract Terraform version from constraints in module
+        run: echo TF_VERSION=$(cat *.tf | ../../hack/ci/hclq_linux_amd64 -r '.terraform.required_version' | head -1 | tr -d ' ') >> $GITHUB_ENV
```

Seems overkill to use a custom tool (which is checked into the repo!) when you can achieve the same with `grep` and `cut`. I would just require modules to define their `terraform` block in `providers.tf` and use the following:

```suggestion
        run: echo TF_VERSION=$(grep "required_version = " providers.tf | cut -d= -f2- | tr -d ' "') >> $GITHUB_ENV
```

---

It seemed really fragile to me and adds an implicit file naming requirement. On the other hand I guess there won't be many more modules so 🤷‍♀️.